integrations/pgvector/src/haystack_integrations/document_stores/pgvector/document_store.py

```diff
@@ -0,0 +1,377 @@
+            https://github.com/pgvector/pgvector?tab=readme-ov-file#hnsw
+        """
+
+        self.connection_string = connection_string
+        self.table_name = table_name
+        self.embedding_dimension = embedding_dimension
+        self.embedding_similarity_function = embedding_similarity_function
+        self.recreate_table = recreate_table
+        self.search_strategy = search_strategy
+        self.hnsw_recreate_index_if_exists = hnsw_recreate_index_if_exists
+        self.hnsw_index_creation_kwargs = hnsw_index_creation_kwargs or {}
+        self.hnsw_ef_search = hnsw_ef_search
+
+        connection = connect(connection_string)
+        connection.autocommit = True
+        self._connection = connection
+
+        # we create a generic cursor and another one that returns dictionaries
+        self._cursor = connection.cursor()
+        self._dict_cursor = connection.cursor(row_factory=dict_row)
+
+        connection.execute("CREATE EXTENSION IF NOT EXISTS vector")
+        register_vector(connection)
+
+        if recreate_table:
+            self.delete_table()
+        self._create_table_if_not_exists()
+
+        if search_strategy == "hnsw":
+            self._handle_hnsw()
+
+    def to_dict(self) -> Dict[str, Any]:
+        return default_to_dict(
+            self,
+            connection_string=self.connection_string,
+            table_name=self.table_name,
+            embedding_dimension=self.embedding_dimension,
+            embedding_similarity_function=self.embedding_similarity_function,
+            recreate_table=self.recreate_table,
+            search_strategy=self.search_strategy,
+            hnsw_recreate_index_if_exists=self.hnsw_recreate_index_if_exists,
+            hnsw_index_creation_kwargs=self.hnsw_index_creation_kwargs,
+            hnsw_ef_search=self.hnsw_ef_search,
+        )
+
+    def _execute_sql(
+        self, sql_query: Query, params: Optional[tuple] = None, error_msg: str = "", cursor: Optional[Cursor] = None
+    ):
+        """
+        Internal method to execute SQL statements and handle exceptions.
+
+        :param sql_query: The SQL query to execute.
+        :param params: The parameters to pass to the SQL query.
+        :param error_msg: The error message to use if an exception is raised.
+        :param cursor: The cursor to use to execute the SQL query. Defaults to self._cursor.
+        """
+
+        params = params or ()
+        cursor = cursor or self._cursor
+
+        try:
+            result = cursor.execute(sql_query, params)
+        except Error as e:
+            self._connection.rollback()
+            raise DocumentStoreError(error_msg) from e
+        return result
+
+    def _create_table_if_not_exists(self):
+        """
+        Creates the table to store Haystack documents if it doesn't exist yet.
+        """
+
+        table_structure_str = ", ".join(f"{name} {dtype} {constraint}" for name, dtype, constraint in TABLE_DEFINITION)
```

Comment:
Why not storing the SQL statement as a string? Something like
```py
CREATE_TABLE_STMT = """
CREATE TABLE IF NOT EXISTS %s (
'id' VARCHAR(128) PRIMARY KEY,
'embedding' VECTOR(%s),
...
);
"""

self._execute_sql(CREATE_TABLE_STMT, params={...}, error_msg="Could not create table in PgvectorDocumentStore")
```

To me this would have several advantages:
- Code is easier to read than this for-loop
- I can go and copy the query to paste it in my pg console for testing
- If you know SQL you know what's going to happen at glance

Reply:
will do!